tests/phpunit/CiviTest/CiviUnitTestCase.php

```diff
@@ -486,6 +486,17 @@ protected function assertPostConditions(): void {
     $this->validateAllContributions();
   }
 
+  /**
+   * Temporary pseudo-poly-fill for php8 to php9 transition.
+   *
+   * @param string $pattern
+   * @param string $string
+   * @param string $message
+   */
+  protected function assertMatchesRegularExpression(string $pattern, string $string, string $message = '') {
+    $this->assertTrue(\preg_match($pattern, $string) > 0, $message);
+  }
```

Comment:
If there's only 1-4 that matter for Civi, then I agree it's simpler to fill these in ourselves.

However:

1. It should be a trait. (That can be used in, say `CiviUnitTestCase` and `Api4TestBase`.)
2. You'll get better assertion-messages from [this technique](https://github.com/totten/git-scan/blob/2022-07-15/tests/GitScan/GitScanTestCase.php#L105)

Reply:
@totten hmm - I think I'd rather treat the test suites differently rather than put a temporary trait in the `Civi` namespace. It bothers me that I put one in recently & I'm hoping to remove it again cos we did make the statement at one point that functions in the \Civi namespace are supported to be called from outside of core (ie are part of the api) - I think that was not necessarily a good thing to say - but our supported surface area seems to be getting much greyer

Reply:
That namespace thing sounds like a longer conversation.

On the implementation of the assertion, I believe is better -- it's as short as your original commit, and it gives the same messaging as my hyperlinked example.

```php
static::assertThat($string, new RegularExpression($pattern), $message);
```

Reply:
Yeah - I don't deeply care - I was expecitng this to be a short-term assert to get us past an issue & just to 'bang something in that does the trick & will last for a few months' - but will close this in favour of your thing